src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedTovStar.cpp

```diff
@@ -44,61 +44,65 @@ void MagnetizedTovStar::pup(PUP::er& p) {
   p | vector_potential_amplitude_;
 }
 
-template <typename DataType>
-tuples::TaggedTuple<hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>>
-MagnetizedTovStar::variables(
-    const tnsr::I<DataType, 3>& coords,
-    tmpl::list<
-        hydro::Tags::MagneticField<DataType, 3, Frame::Inertial>> /*meta*/,
-    const RadialVariables<DataType>& radial_vars) const {
+namespace magnetized_tov_detail {
+template <typename DataType, StarRegion Region>
+void MagnetizedTovVariables<DataType, Region>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 3>*> magnetic_field,
+    const gsl::not_null<Cache*> cache,
+    hydro::Tags::MagneticField<DataType, 3> /*meta*/) const {
   const size_t num_pts = get_size(get<0>(coords));
-  auto magnetic_field =
-      make_with_value<tnsr::I<DataType, 3, Frame::Inertial>>(num_pts, 0.0);
-  using std::max;
-  const auto sqrt_det_spatial_metric =
-      get<gr::Tags::SqrtDetSpatialMetric<DataType>>(variables(
-          coords, tmpl::list<gr::Tags::SqrtDetSpatialMetric<DataType>>{},
-          radial_vars));
+  const auto& pressure_profile =
+      get(cache->get_var(*this, hydro::Tags::Pressure<DataType>{}));
+  const auto& dr_pressure_profile = get(cache->get_var(
+      *this,
+      RelativisticEuler::Solutions::tov_detail::Tags::DrPressure<DataType>{}));
+  const auto& sqrt_det_spatial_metric =
+      get(cache->get_var(*this, gr::Tags::SqrtDetSpatialMetric<DataType>{}));
   for (size_t i = 0; i < num_pts; ++i) {
-    const double pressure = get_element(get(radial_vars.pressure), i);
-    if (LIKELY(get_element(radial_vars.radial_coordinate, i) > 1.0e-16)) {
-      if (pressure < cutoff_pressure_) {
+    const double pressure = get_element(pressure_profile, i);
+    if (LIKELY(get_element(radius, i) > 1.0e-16)) {
+      if (pressure < cutoff_pressure) {
+        get_element(get<0>(*magnetic_field), i) = 0.;
```

Comment:
0. -> 0.0 and throughout the rest of the file

Reply:
squashed this in